### PR TITLE
marktext: 0.16.2 -> 0.16.3

### PR DESCRIPTION
**Pull request by eduarrrd**
Add .desktop file, icons in the process.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by eduarrrd** [comment]
@nh2 , let me know if you have concerns.

File: pkgs/applications/misc/marktext/default.nix [hunk 1]
```diff
@@ -28,8 +33,16 @@ appimageTools.wrapType2 rec {
     p.xorg.libxkbfile
   ];
 
-  # Strip version from binary name.
-  extraInstallCommands = "mv $out/bin/${name} $out/bin/${pname}";
+  extraInstallCommands = ''
+    # Strip version from binary name.
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/marktext/default.nix:37]:
```suggestion
    # rename binary to not contain version
```
striping a binary usually means something else.

File: pkgs/applications/misc/marktext/default.nix [hunk 2]
```diff
@@ -2,16 +2,21 @@
 
 let
   pname = "marktext";
-  version = "v0.16.2";
-in
-appimageTools.wrapType2 rec {
+  version = "v0.16.3";
   name = "${pname}-${version}-binary";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/marktext/default.nix:6]:
```suggestion
  version = "v0.16.3";
```
all appimage things now support pname+version.

File: pkgs/applications/misc/marktext/default.nix [hunk 3]
```diff
@@ -2,16 +2,21 @@
...
-    sha256 = "0ivf9lvv2jk7dvxmqprzcsxgya3617xmx5bppjvik44z14b5x8r7";
+    sha256 = "0s93c79vy2vsi7b6xq4hvsvjjad8bdkhl1q135vp98zmbf7bvm9b";
   };
 
+  appimageContents = appimageTools.extractType2 {
+    inherit name src;
+  };
+in
+appimageTools.wrapType2 rec {
+  inherit name src;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/marktext/default.nix:18]:
```suggestion
  inherit pname src;
```

File: pkgs/applications/misc/marktext/default.nix (hunk 1 above)
Comment (eduarrrd) [reply pkgs/applications/misc/marktext/default.nix:37]: Definitely, but the comment doesn't talk about stripping the binary though. Looks like a regular string operation, e.g.

 ```python
>>> "binary-version".strip("-version")
'binary'
```

I'm fine changing this if you insist.

File: pkgs/applications/misc/marktext/default.nix (hunk 2 above)
Comment (eduarrrd) [reply pkgs/applications/misc/marktext/default.nix:6]: Cool, didn't know that. I would be happy to change this but given the recency of you introducing support for this ... that breaks backporting, no?

Comment (SuperSandro2000) [reply pkgs/applications/misc/marktext/default.nix:6]: > that breaks backporting

yeah, probably

File: pkgs/applications/misc/marktext/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/marktext/default.nix:37]: yeah, I know what you mean but striping something from a binary in a build process usually means removing debugging symbols. I just thought the comment is not that optimal.

**Pull request merged**